src/geom/line.jl

```diff
@@ -131,7 +131,7 @@ function render(geom::LineGeometry, theme::Gadfly.Theme, aes::Gadfly.Aesthetics)
                       svgclass(classes))
 
     elseif length(aes.color) == 1 &&
-            !(isa(aes.color, IndirectArray) && length(levels(aes.color)) > 1)
+            !(isa(aes.color, IndirectArray) && length(filter(!ismissing, aes.color.values)) > 1)
```

Comment:
Could also use `count(!ismissing, aes.color.values)` to avoid an allocation. Same below.